### PR TITLE
Allow diagrams under all elements

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [X] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Diagrams can only be put under a Package. This is an okay position, but we got requests to allow for diagrams under other model elements as well.

Issue Number: #676

### What is the new behavior?

Diagrams can be placed under any element. 

This has the advantage that we can also clean up the Core model. So far, it was tightly coupled to the UML model (by means of NamedElement and PackageableElement.

image

(the old core model can be [found in the docs](https://gaphor.readthedocs.io/en/latest/models/core.html).

### Does this PR introduce a breaking change?
- [X] Yes
- [ ] No

### CODE REVIEW

**Review by danyeaw — COMMENTED** [review]
This looks good! I did notice a couple of things that I didn't expect when I gave it a spin:

1. From the TreeView, if you right click on an element and select New Diagram, it creates it under the main model package. I was expecting it to nest the diagram under the element that is selected.
2. If you create a diagram and then drag it under an element, then create an element on the diagram, the new element isn't automatically nested under the diagram owner element.

File: gaphor/UML/classes/containmentconnect.py [hunk 1]
```diff
@@ -67,14 +71,16 @@ def disconnect(self, handle):
 
         if hct and oct:
             if hct.subject in oct.subject.ownedElement:
-                assert isinstance(hct.subject, (UML.Type, UML.Package, UML.Diagram))
+                assert isinstance(hct.subject, (UML.Type, UML.Package))
                 if isinstance(hct.subject, UML.Classifier):
                     del hct.subject.nestingClass
-                hct.subject.package = hct.diagram.package
+                # TODO: look up package
```
[the remark this answers is not in the document]
Comment (amolenaar) [reply gaphor/UML/classes/containmentconnect.py:77]: I think I missed this one.

**Comment by amolenaar** [comment]
> This looks good! I did notice a couple of things that I didn't expect when I gave it a spin:
> 
>     1. From the TreeView, if you right click on an element and select New Diagram, it creates it under the main model package. I was expecting it to nest the diagram under the element that is selected.

I think this still needs fixing, as well as the "create diagram" button in the header.

>     2. If you create a diagram and then drag it under an element, then create an element on the diagram, the new element isn't automatically nested under the diagram owner element.

In many cases this is not possible. E.g. when you create a class in a diagram under an Interaction. I tried to stay as close to the original behavior as possible, to avoid introducing new errors.

File: gaphor/ui/namespacemodel.py [hunk 2]
```diff
@@ -273,7 +273,11 @@ def do_drag_data_received(self, dest_path, selection_data):
                 if ns is element:
                     log.info("Can not create a cycle")
                     return False
-                ns = ns.namespace
```
Comment (amolenaar) [inline gaphor/ui/namespacemodel.py:274]: This should have been the `owner` (`namespace` is a subset of `owner`).

**Pull request merged**